experimental/errorsource/error_source_middleware.go

```diff
@@ -0,0 +1,35 @@
+package errorsource
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// ErrorSourceMiddlewareName is the middleware name used by ErrorSourceMiddleware.
+const ErrorSourceMiddlewareName = "ErrorSource"
+
+var (
+	errors = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "plugins",
+		Name:      "datasource_error_total",
+		Help:      "Total number of times a plugin errored",
+	}, []string{"error_source"})
+)
+
+// ErrorSourceMiddleware captures error source metric
+func ErrorSourcenMiddleware() httpclient.Middleware {
+	return httpclient.NamedMiddlewareFunc(ErrorSourceMiddlewareName, func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+		return httpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			res, err := next.RoundTrip(req)
+			if res != nil && res.StatusCode >= 400 {
+				errorSource := backend.GetErrorSource(res.StatusCode)
+				errors.WithLabelValues(string(errorSource)).Inc()
```

Comment:
Noticed you added metrics here now instead - How would you use this? There's only a status source label and nothing else? I imagine you're trying to do part of Grafana changes for plugin request metrics in the [epic](https://github.com/grafana/grafana/issues/75761)?

Reply:
yes trying to do part of the grafana changes until we have something there.  also, not sure how current/older versions of grafana will capture this ( or if we care about older versions ).

Reply:
Will showed how to add a custom metric in plugins that will get stored in prometheus.  I assumed this would capture a custom metric of "datasource_error_total" with an  error_source label?

Reply:
removed